components/micro-gateway-core/src/main/ballerina/src/gateway/jwt_generator/jwt_gen_util.bal

```diff
@@ -295,14 +285,47 @@ function getGeneratedTokenExpTimeFromCache(string cacheKey, string jwtToken) ret
     }
 }
 
-# To update the custom claims of jwt payload with principal's custom claims. This is required to pass the mapped claims.
-# + payload - JWT Payload
-# + return - modified payload with principal's claims
-function updateCustomClaimsUsingPrincipal(jwt:JwtPayload payload) returns jwt:JwtPayload {
-    runtime:Principal? principal = runtime:getInvocationContext()?.principal;
-    if (principal is runtime:Principal) {
-            map<any>? principalClaims = principal?.claims;
-            payload.customClaims = <(map<json>)> principalClaims;
+# Generate the backend JWT token and set to the header of the outgoing request.
+#
+# + credential - Credential
+# + authContext - Authentication Context
+# + req - The `Request` instance.
+# + enabledJWTGenerator - state of jwt generator
+# + classLoaded - whether the class is loaded successfully
+# + enabledCaching - jwt generator caching enabled
+# + skewTime - skew time to backend
+# + issuer - The jwt issuer who issued the token and comes in the iss claim.
+# + remoteUserClaimRetrievalEnabled - true if remoteUserClaimRetrieval is enabled
+# + isJWT - `true` if the user is authenticated using jwt token
+# + return - Returns `true` if the token generation and setting the header completed successfully
+# or the `AuthenticationError` in case of an error.
+public function generateAndSetBackendJwtHeader(string credential,
+                                                http:Request req,
+                                                AuthenticationContext authContext,
```

Comment:
Is there a usage of the authContext in this method? We can remove it.

Reply:
Thanks for pointing out.

Reply:
fixed with https://github.com/wso2/product-microgateway/pull/1345/commits/337f92d8f381cc6a994235a0a15b8671a492c263